lib/rucio/core/replica.py

```diff
@@ -569,9 +569,22 @@ def _resolve_dids(dids, unavailable, ignore_availability, all_states, session):
                                    models.DataIdentifier.name == did['name']))
 
     if did_clause:
-        for scope, name, did_type in session.query(models.DataIdentifier.scope,
-                                                   models.DataIdentifier.name,
-                                                   models.DataIdentifier.did_type).with_hint(models.DataIdentifier, "INDEX(DIDS DIDS_PK)", 'oracle').filter(or_(*did_clause)):
+        for scope, name, did_type, constituent in session.query(models.DataIdentifier.scope,
+                                                                models.DataIdentifier.name,
+                                                                models.DataIdentifier.did_type,
+                                                                models.DataIdentifier.constituent)\
+                                                         .with_hint(models.DataIdentifier, "INDEX(DIDS DIDS_PK)", 'oracle')\
+                                                         .filter(or_(*did_clause)):
+            if constituent:
+                # file is a constituent, resolve to any parent archive
+                archive = session.query(models.ConstituentAssociation.scope,
+                                        models.ConstituentAssociation.name)\
+                                 .filter(models.ConstituentAssociation.child_scope == scope,
+                                         models.ConstituentAssociation.child_name == name)\
+                                 .with_hint(models.ConstituentAssociation, "INDEX(ARCHIVE_CONTENTS ARCH_CONTENTS_PK)", 'oracle')\
```

Comment:
Did you check how I define the PK index with child_scope, child_name first to avoid range scans ?  I think I did to avoid the secondary reverse index but better to check.

Reply:
child comes first, yes. But as a file can be registered to multiple zips, we have to get all of them.

Reply:
yes, I think I assumed that getting parent zip files was a more popular use case than getting archive content. Better to keep an eye on it wrt wfms access patterns.

Reply:
in my tests it was ok, let's see how it scales, i'd like to avoid another index on replicas :-D

Reply:
The concerned table is archive_content, not the replicas table. No?

Reply:
yes, of course, i typed too quickly